mlflow/utils/requirements_utils.py

```diff
@@ -267,7 +267,27 @@ def _capture_imported_modules(model_uri, flavor):
             return f.read().splitlines()
 
 
-_MODULES_TO_PACKAGES = None
+def _gen_model_package_maps():
+    # Note `importlib_metada.packages_distributions` only captures packages installed into
+    # Python’s site-packages directory via tools such as pip:
+    # https://importlib-metadata.readthedocs.io/en/latest/using.html#using-importlib-metadata
+    _modules_to_packages = importlib_metadata.packages_distributions()
+
+    # In Databricks, `_MODULES_TO_PACKAGES` doesn't contain pyspark since it's not installed
+    # via pip or conda. To work around this issue, manually add pyspark.
+    if is_in_databricks_runtime():
+        _modules_to_packages.update({"pyspark": ["pyspark"]})
+
+    _packages_to_modules = {}
+    for module, pkg_list in _modules_to_packages.items():
+        for pkg_name in pkg_list:
+            _packages_to_modules[pkg_name] = module
+
+    return _modules_to_packages, _packages_to_modules
+
+
+_MODULES_TO_PACKAGES, _PACKAGES_TO_MODULES = _gen_model_package_maps()
```

Comment:
How long does it take for `_gen_model_package_maps` to complete?

Reply:
If `_gen_model_package_maps` takes long, `import mlflow` takes long.

Reply:
change back to lazy way